src/averageAges.js

```diff
@@ -54,8 +68,23 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
-function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+function calculateAverageAgeDiff(people, onlyWithSon = false) {
+  const children = people.filter(child => {
+    return people.some(mother => child.mother === mother.name)
+      && (onlyWithSon
+        ? child.sex === 'm'
+        : true);
+  });
```

Comment:
```suggestion
  const children = people.filter(child => (
    people.some(mother => child.mother === mother.name)
      && (!onlyWithSon || child.sex === 'm')
  ));
```

You can do a direct return for the arrow function. And create a more simple check for `onlyWithSon`.